great_expectations/datasource/data_connector/asset/asset.py

```diff
@@ -13,15 +13,19 @@ class Asset:
     def __init__(
         self,
         name: str,
-        base_directory: str = None,
-        glob_directive: str = None,
-        pattern: str = None,
+        base_directory: Optional[str] = None,
+        glob_directive: Optional[str] = None,
+        pattern: Optional[str] = None,
         group_names: List[str] = None,
-        bucket: str = None,
-        prefix: str = None,
-        delimiter: str = None,
+        # S3
+        bucket: Optional[str] = None,
+        prefix: Optional[str] = None,
         max_keys: int = None,
-        batch_spec_passthrough: dict = None,
+        # Azure
+        container: Optional[str] = None,
+        name_starts_with: Optional[str] = None,
+        delimiter: Optional[str] = None,  # Both S3/Azure
```

Comment:
As in "# Both S3/Azure" above `delimiter`?

Reply:
Ah nevermind I think I know what you mean! I now realize that ALL the other fields are actually shared by both of them (not just `delimiter`).

Here's an updated view:
```
    def __init__(
        self,
        name: str,
        base_directory: Optional[str] = None,
        glob_directive: Optional[str] = None,
        pattern: Optional[str] = None,
        group_names: List[str] = None,
        delimiter: Optional[str] = None,
        batch_spec_passthrough: Optional[dict] = None,
        # S3
        bucket: Optional[str] = None,
        prefix: Optional[str] = None,
        max_keys: int = None,
        # Azure
        container: Optional[str] = None,
        name_starts_with: Optional[str] = None,
    ):
        self._name = name
        self._base_directory = base_directory
        self._glob_directive = glob_directive
        self._pattern = pattern
        # Note: this may need to become a nested object to accommodate sorters
        self._group_names = group_names
        self._delimiter = delimiter
        self._batch_spec_passthrough = batch_spec_passthrough or {}

        # S3
        self._bucket = bucket
        self._prefix = prefix
        self._max_keys = max_keys

        # Azure
        self._container = container
        self._name_starts_with = name_starts_with

```

Reply:
I believe `delimiter` is used in both Azure and S3, but not for the Filesystem connectors; if that's the case, I'd pull it to its own group, with the comment you had before (# both S3 and Azure) or similar.

Reply:
ah okay that's a good catch thank you for calling that out!

I've made a quick update; mind letting me know if that fits what you're envisioning?

Reply:
Yup, newest version looks good, thanks!